spec/unit/classes/init_spec.rb

```diff
@@ -3,28 +3,20 @@
 describe 'puppetdb', :type => :class do
   ttl_args = ['node_ttl','node_purge_ttl','report_ttl']
 
-  context 'on a supported platform' do
-    let(:facts) do
-      {
-        :osfamily => 'Debian',
-        :puppetversion => Puppet.version,
-        :operatingsystem => 'Debian',
-        :operatingsystemrelease => '6.0',
-        :kernel => 'Linux',
-        :concat_basedir => '/var/lib/puppet/concat',
-        :lsbdistid => 'Debian',
-        :lsbdistcodename => 'foo',
-        :id => 'root',
-        :path => '/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin',
-        :selinux => false,
-        :iptables_persistent_version => '0.5.7',
-      }
-    end
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
+      let(:facts) do
+        facts.merge({
```

Comment:
The selinux fact changes in different tests, but the iptables fact never changes.